### PR TITLE
Fix handling of 404 when fetching downloads (no stats yet)

### LINKED ISSUE #150: Fix 404 handling in npm downloads API
The new downloads API was recently released and the 404 detection needs to be updated: https://github.com/npms-io/npms-analyzer/blob/master/lib/analyze/collect/npm.js#L42

### PR DESCRIPTION
**Pull request by satazor**
Fixes #150

### CODE REVIEW

**Comment by satazor** [comment]
//cc @bcoe

File: lib/analyze/collect/npm.js [hunk 1]
```diff
@@ -30,16 +30,9 @@ function fetchDownloads(name) {
...
-            throw new Error(response.body.error);
-        }
-
-        return response.body.downloads;
-    })
-    // Check if the error is "no stats" and return an empty array
-    .catch((err) => /no stats for this package/i.test(err.message), () => [])
+    .then((res) => res.body.downloads)
+    // Check if there is no stats yet
+    .catch({ statusCode: 404 }, () => [])
```
Comment (bcoe) [inline lib/analyze/collect/npm.js]: cool, this only works with Bluebird promises right?

Comment (satazor) [inline lib/analyze/collect/npm.js]: got throws an error that has `statusCode` and bluebird allows to do `catch` this way.

Comment (satazor) [inline lib/analyze/collect/npm.js]: yes this only works because we are using bluebird globally.

**Comment by satazor** [comment]
Going to deploy.

**Pull request merged**